kge/util/dump.py

```diff
@@ -113,90 +113,174 @@ def _add_dump_trace_parser(subparsers_dump):
+        ),
+    )
+    parser_dump_trace.add_argument(
+        "--keys",
+        "-k",
+        nargs="*",
+        type=str,
+        help=(
+            "A list of 'key' entries (separated by space). Each 'key' has form"
+            " 'new_key_name=key_name' or 'key_name'. This will add columns as in the"
+            " --keysfile option. When only 'key_name' is provided, it is also used as"
+            " column name."
+        ),
+    )
     parser_dump_trace.add_argument(
         "--checkpoint",
         default=False,
         action="store_const",
         const=True,
         help=(
-            "If source is a path to a job folder and --checkpoint is set the best "
-            "(if present) or last checkpoint will be used to determine the job_id"
+            "If source is a path to a job folder and --checkpoint is set, the best"
+            " (if present) or last checkpoint will be used to determine the job_id from"
+            " where the tracefile will be processed backwards."
         ),
     )
-
     parser_dump_trace.add_argument(
         "--job_id",
         default=False,
         help=(
-            "Specifies the training job id in the trace "
-            "from where to start processing backward"
+            "Specifies the training job id in the trace"
+            " from where to start processing the tracefile backward."
         ),
     )
-
     parser_dump_trace.add_argument(
         "--max_epoch",
         default=False,
         help=(
-            "Specifies the epoch in the trace"
-            "from where to start processing backwards"
+            "Specifies the max epoch number in the tracefile"
+            " from where to start processing backwards. Can not be used with --truncate"
+            " when a checkpoint is provided."
         ),
     )
-
     parser_dump_trace.add_argument(
         "--truncate",
         default=False,
         action="store_const",
         const=True,
         help=(
-            "If a checkpoint is used (by providing one explicitly as source or by "
-            "using --checkpoint), --truncate will define the max_epoch to process as"
-            "specified by the checkpoint"
+            "When a checkpoint is used (by providing one explicitly as source or by"
+            " using --checkpoint), --truncate will define the max_epoch number to"
+            " process as provided by the checkpoint. Can only be used when a checkpoint"
+            " is provided and it cannot be used together with --max_epoch."
         ),
     )
-
-    for argument in [
-        "--train",
-        "--valid",
-        "--test",
-        "--search",
+    parser_dump_trace.add_argument(
         "--yaml",
+        action="store_const",
+        const=True,
+        default=False,
+        help="Instead of using a CSV file with a subset of entries, dump the full"
+             " trace file. Additional entries from the config can be added with the"
```

Comment:
Still not true.

Reply:
Please make a suggestion.

Reply:
It's not the full trace file, but the same subset that would have been used for the CSV. So simply: "Dump YAML instead of CSV"?

Reply:
I'm not sure if I understand you correctly. The --csv option only dumps the default keys whereas the --yaml option directly dumps the complete  trace entry. It does not filter out anything, which is how I was instructed. So what do you mean with the "same subset"

Reply:
Yes, the complete trace entry for the subset of selected trace entries. Not all trace entries in the trace file.

Reply:
Ok, we mean the same. Complete trace is indeed misleading.